public/app/types/unified-alerting.ts

```diff
@@ -157,6 +172,30 @@ export interface RuleGroupIdentifier {
   groupName: string;
 }
 
+export interface GrafanaNamespaceIdentifier {
+  uid: string;
+}
+
+export interface DataSourceNamespaceIdentifier {
+  name: string;
+}
+
+export interface GrafanaRuleGroupIdentifier {
+  rulesSource: { uid: typeof GrafanaRulesSourceSymbol; name: typeof GRAFANA_RULES_SOURCE_NAME };
+  groupName: string;
+  namespace: GrafanaNamespaceIdentifier;
+  groupOrigin: 'grafana';
```

Comment:
is this `groupOrigin` redundant? I understand that depends on the rulesSource, isn't it?

Reply:
I think we can use maybe a util function to get it, instead of adding this field in here?

Reply:
ok! nice!!! great solution!